src/Orleans.Runtime/Services/GrainServicesSiloBuilderExtensions.cs

```diff
@@ -0,0 +1,35 @@
+﻿using Microsoft.Extensions.DependencyInjection;
+using Orleans.CodeGeneration;
+using Orleans.Core;
+using Orleans.Runtime;
+using Orleans.Services;
+using System;
+using System.Linq;
+
+namespace Orleans.Hosting
+{
+    public static class GrainServicesSiloBuilderExtensions
+    {
+        /// <summary>
+        /// Registers an application grain service to be started with the silo.
+        /// </summary>
+        public static ISiloHostBuilder AddGrainService<T>(this ISiloHostBuilder builder)
+            where T : class, IGrainService
```

Comment:
Since we have a hard cast to `GrainService` below, I think we should use `GrainService` as the constraint here